doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,148 @@
+
+Run the following command:
+
+```bash
+cabal install cardano-cli
+cabal install cardano-node
+./scripts/babbage/mkfiles.sh
+```
+
+To start your babbage cluster, you need to run the `example/run/all.sh` shell script.
+The remainder of this guide provides a brief walkthrough of the [shell script example](scripts/plutus/example-reference-script-usage.sh) that automatically creates a reference script and spends the utxo at
+the reference script's corresponding script address.
+
+#### Creating a reference script at a transaction output and
+#### sending ada to the script address (with a datum)
+
+In order to use a reference script, we must first create this script at a particular transaction output.
+
+```bash
+cardano-cli transaction build \
+  --babbage-era \
+  --cardano-mode \
+  --testnet-magic 42 \
+  --change-address "$utxoaddr" \
+  --tx-in "$txin" \
+  --tx-out "$utxoaddr+$lovelace" \
+  --tx-out "$plutusscriptaddr+$lovelace" \
+  --tx-out-datum-hash "$scriptdatumhash" \
+  --tx-out "$dummyaddress+$lovelaceattxindiv3" \
+  --tx-out-reference-script-file "$plutusscriptinuse" \
+  --protocol-params-file "$WORK/pparams.json" \
+  --out-file "$WORK/create-datum-output.body"
+```
+
+The following should be noted about this build command:
+
+Firstly, we are sending ada to the plutus script address along with a datum hash. This is reflected in the following lines:
+
+```bash
+...
+--tx-out "$plutusscriptaddr+$lovelace" \
+--tx-out-datum-hash "$scriptdatumhash" \
+...
+```
+
+We have seen this before in the [plutus-spending-script-example.md](doc/reference/plutus/plutus-spending-script-example.md).
+
+Secondly, we are creating a reference script at a tx output, which cannot be the same as the reference script address of our choice:
+
+```bash
+...
+--tx-out "$dummyaddress+$lovelaceattxindiv3" \
+--tx-out-reference-script-file "$plutusscriptinuse" \
+...
+```
+
+Specifying the `--reference-script-file` after the `--tx-out` option will construct a transaction that creates a reference script at that transaction output.
+
+We sign and then submit as usual:
+
+```bash
+cardano-cli transaction sign \
+  --tx-body-file $WORK/create-datum-output.body \
+  --testnet-magic "$TESTNET_MAGIC" \
+  --signing-key-file $UTXO_SKEY \
+  --out-file $WORK/create-datum-output.tx
+```
+
+
+#### Spending ada at the script address
+
+Now that there is ada at our script address, we must construct the appropriate transaction in order to spend it.
+Because we are using the `build` command, we should only note the following:
+
+`$plutusutxotxin` - This is the tx input that sits at the Plutus script address (NB: It has a datum hash).
+`tx-in-reference` - This specifies the reference input you are using to witness a transaction input.
+`plutus-script-v2`- This specifies the version of the reference script at the reference input. There are no restrictions on the type of script we can have at a reference input.
```

Comment:
> There are no restrictions on the type of script we can have at a reference input.

Did we want to remove this altogether, or explain that V1 scripts can be referenced but not executed?

Reply:
Yeah let's remove that. I will add a separate cli parser for "stand alone" reference inputs that don't witness anything